### PR TITLE
ENH: stats.normaltest/jarque_bera: add array-API support

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Toward gh-20544

#### What does this implement/fix?
Adds array API support to `stats.normaltest` and `stats.jarque_bera`.

#### Additional information
Draft to let gh-20085 merge first.
Will also address https://github.com/scipy/scipy/pull/20085#issuecomment-2118280655 if we figure out what was going on there.

Next PR will rewrite all these tests.

### CODE REVIEW

File: scipy/stats/_stats_py.py [hunk 1]
```diff
@@ -2001,9 +2001,16 @@ def normaltest(a, axis=0, nan_policy='propagate'):
     """
     s, _ = skewtest(a, axis)
     k, _ = kurtosistest(a, axis)
-    k2 = s*s + k*k
+    Z = s*s + k*k
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:2004]:
```suggestion
    k2 = s*s + k*k
```

File: scipy/stats/_stats_py.py [hunk 2]
```diff
@@ -2001,9 +2001,16 @@ def normaltest(a, axis=0, nan_policy='propagate'):
...
 
-    return NormaltestResult(k2, distributions.chi2.sf(k2, 2))
+    xp = array_namespace(Z)
+    Z_np = np.asarray(Z)
+    pvalue = distributions.chi2.sf(Z_np, 2)
+    pvalue = xp.asarray(pvalue, dtype=Z.dtype)
+    Z = Z[()] if Z.ndim == 0 else Z
+    pvalue = pvalue[()] if pvalue.ndim == 0 else pvalue
+
+    return NormaltestResult(Z, pvalue)
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:2013]:
```suggestion
    xp = array_namespace(k2)
    k2_np = np.asarray(k2)
    pvalue = distributions.chi2.sf(k2_np, 2)
    pvalue = xp.asarray(pvalue, dtype=k2.dtype)
    k2 = k2[()] if k2.ndim == 0 else k2
    pvalue = pvalue[()] if pvalue.ndim == 0 else pvalue

    return NormaltestResult(k2, pvalue)
```

File: scipy/stats/tests/test_stats.py [hunk 3]
```diff
@@ -6308,31 +6310,22 @@ def test_input_validation(self):
 
 
 class TestJarqueBera:
-    def test_jarque_bera_stats(self):
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:6311]: Replaced with a stronger test.

**Review by j-bowhay — APPROVED** [review]
Thanks LGTM

**Pull request merged**